corehq/apps/users/management/commands/remove_mobile_permissions.py

```diff
@@ -0,0 +1,36 @@
+from django.core.management.base import BaseCommand
+
+from corehq.apps.domain.models import Domain
+from corehq.apps.users.models_role import RolePermission, UserRole
+from corehq.toggles import RESTRICT_MOBILE_ACCESS
+
+
+class Command(BaseCommand):
+    help = "Removes the mobile access permission from roles not in Domains with" \
+           " the RESTRICT_MOBILE_ACCESS feature flag"
+
+    def handle(self, **options):
+        num_domains_with_their_roles_changed = 0
+        domains_with_restriction_on = RESTRICT_MOBILE_ACCESS.get_enabled_domains()
+        for domain in Domain.get_all_names():
```

Comment:
there are a lot of domains. It would probably be a much faster to get the list of relevant domains from the roles table:

```python
domains = UserRole.objects.filter(
rolepermission__permission_fk__value='access_mobile_endpoints').values_list('domain', flat=True)
```

Reply:
Awesome, clearly a much better way to filter down. Added, thanks for this.